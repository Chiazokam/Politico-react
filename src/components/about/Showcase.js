import React, {Component} from 'react';
import '../../styles/about/showcase.scss';

class Showcase extends Component {
  render() {
    return (
      <section>
        <div id="main-about">
          <p className="p-start">
            It’s a political season. Get into the mood of the
            season and help build our future, one tenure at a time.
            Politico enables citizens give their mandate to politicians running for different government offices
            while building trust in the process through transparency.It’s a political season. Get into the mood of the
            season and help build our future, one tenure at a time.
          </p>
          <p>
            Join the band of democrats making the nation's decision.
          </p>
          <p>
            Politico is fast and easy to use. Get on in and no second would be wasted!
          </p>
        </div>
      </section>
    )
  }
}

export default Showcase;
