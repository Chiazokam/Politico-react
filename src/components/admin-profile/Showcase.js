import React, { Component } from 'react';

class Showcase extends Component {
  render() {
    return (
      <div id="main-profile">
        <div className="flex-container">
          <div className="card">
              <a href="/parties/new"><button type="button" className="btn btn-trans">Create a Political Party</button></a>
          </div>  
          <div className="card">
              <a href="/parties"><button type="button" className="btn btn-trans">View Political Parties</button></a>
          </div> 
          <div className="card">
              <a href="/offices/new"><button type="button" className="btn btn-trans">Create a Political Office</button></a>
          </div> 
          <div className="card">
              <a href="#"><button type="button" className="btn btn-trans">View Political Offices</button></a>
          </div> 
        </div>
      </div>
    )
  }
}

export default Showcase;
