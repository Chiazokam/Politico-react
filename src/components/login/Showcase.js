import React, {Component} from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSpinner } from '@fortawesome/free-solid-svg-icons';
import { Container, FormField, Url, Button } from '../global';
import Modal from './Reset-password-modal';
import { showModal, closeModal } from '../../actions';
import '../../styles/login/login.scss';
import { signinUser } from '../../actions';

class Showcase extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      errors: {},
      redirect: false,
      decode: false,
      submit: false
    };
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const user = {
      email: this.state.email,
      password: this.state.password,
    };
    this.props.dispatch(signinUser(user));
  }

  clearField = (e) => {
    this.setState({ errors: '' })
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value});
  }

  showModal = () => {
    this.props.dispatch(showModal());
  }

  closeModal = () => {
    this.props.dispatch(closeModal());
  }

  render() {
    if (this.props.redirect && this.props.isAdmin) return <Redirect to='/admin-profile' />;
    else if (this.props.redirect && !this.props.isAdmin) {
      return <Redirect to='/user-profile' />;
    }
    return (
      <div id="main-signin">
        <Container>
        { this.props.isModalOpen && <Modal onClick={this.closeModal}/> }            
          <p className="form-text">Sign In</p>
          <form onSubmit={this.handleSubmit.bind(this)}>
            <div className="error">{this.props.errors.message}</div>
            <FormField 
              className="form-field"
              error={this.props.errors.email}
              type="email"
              onChange={this.handleChange}
              onFocus={this.clearField}
              name="email"
              value={this.state.email}
              placeholder="Email Address"
            />

            <FormField 
              className="form-field password-field"
              error={this.props.errors.password}
              type="password"
              onChange={this.handleChange.bind(this)}
              onFocus={this.clearField.bind(this)}
              name="password"
              value={this.state.password}
              placeholder="Password"
            />
                                            
            <p id="forgotPass" onClick={this.showModal.bind(this)}>Forgot Password?</p>
            <Button
              className="btn btn-colored btn-signin btn-dark">
              { this.props.submit && <FontAwesomeIcon 
                  icon={ faSpinner }
                  spin
                /> }
                &nbsp; Submit
            </Button>
          </form>
        </Container>
       </div>
    )
  }
}

const mapStateToProps = state => { 
  return {
    isModalOpen: state.global.isModalOpen,
    submit: state.auth.submit,
    errors: state.auth.loginErrors,
    redirect: state.auth.redirect,
    email: state.auth.email,
    password: state.auth.password,
    isAdmin: state.auth.isAdmin
   };
  }

export default connect(mapStateToProps)(Showcase);
 