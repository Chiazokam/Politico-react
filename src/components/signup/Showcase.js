import React, {Component} from 'react';
import axios from 'axios';
import { Redirect } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSpinner } from '@fortawesome/free-solid-svg-icons';
import '../../styles/signup/signup.scss';
import '../../styles/global/form.scss';
import '../../styles/global/container.scss';
import { Container, FormField, Url, Button } from '../global';

class Showcase extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstname: '',
      lastname: '',
      othername: '',
      email: '',
      phone: '',
      passportUrl: '',
      password: '',
      errors: {},
      redirect: false,
      submit: false
    };
  }

  handleSubmit = (e) => {
   e.preventDefault();
   this.setState({
     errors: {},
     submit: true,
    });
    const user = {
      firstname: this.state.firstname,
      lastname: this.state.lastname,
      othername: this.state.othername,
      email: this.state.email,
      phone: this.state.phone,
      passportUrl: this.state.passportUrl,
      password: this.state.password
    };
    axios.post(`${Url.herokuUrl}/auth/signup`, user)
    .then((response) => this.setState({
        redirect: true,
        submit: false,
      }))
    .catch((error) => {
      this.setState({
        errors: error.response.data.error,
        submit: false,
      })
    })
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value})
  } 

  render() {
    if (this.state.redirect) {
      return <Redirect to='/login' />;
    }
    return (
      <div className="main-signup">
          <Container>

          <p className="form-text">Sign Up</p>
          <form onSubmit={this.handleSubmit.bind(this)}>
            <FormField 
              className="form-field" 
              error={this.state.errors.firstname}
              type="text"
              name="firstname"
              onChange={this.handleChange.bind(this)}
              value={this.state.firstname} 
              placeholder="First Name"/>

            <FormField
              className="form-field"
              error={this.state.errors.lastname}
              type="text"
              name="lastname"
              onChange={this.handleChange.bind(this)}
              value={this.state.lastname} 
              placeholder="Last Name"/>

            <FormField
              className="form-field"
              error={this.state.errors.othername}
              type="text"
              name="othername"
              onChange={this.handleChange.bind(this)}
              value={this.state.othername} 
              placeholder="Other Name"/>
          
            <FormField
              className="form-field"
              error={this.state.errors.email}
              type="text"
              name="email"
              onChange={this.handleChange.bind(this)}
              value={this.state.email} 
              placeholder="Email Address"/>

            <FormField
              className="form-field"
              error={this.state.errors.phone}
              type="text"
              name="phone"
              onChange={this.handleChange.bind(this)}
              value={this.state.phone} 
              placeholder="Phone Number"/>

            <FormField
              className="form-field"
              error={this.state.errors.passportUrl}
              type="text"
              name="passportUrl"
              onChange={this.handleChange.bind(this)}
              value={this.state.passportUrl} 
              placeholder="Passport Url"/>

            <FormField
              className="form-field"
              error={this.state.errors.password}
              type="password"
              name="password"
              onChange={this.handleChange.bind(this)}
              value={this.state.password} 
              placeholder="Password"/>
                                            
            <Button
              className="btn btn-colored btn-signup btn-dark">
              { this.state.submit && <FontAwesomeIcon 
                  icon={ faSpinner }
                  spin
                /> }
                &nbsp; Submit
            </Button>
          </form>
          </Container>
        </div> 
    )
  }
} 

export default Showcase;
