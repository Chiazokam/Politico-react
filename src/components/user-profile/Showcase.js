import React, { Component } from 'react';

class Showcase extends Component {
  render() {
    return (
      <div id="main-profile">
      <h2 className='profile-text'>View Election Records</h2>
        <div className="flex-container">
          <div className="card">
              <a href="/parties"><button type="button" className="btn btn-trans">View All Political Parties</button></a>
          </div>  
          <div className="card">
              <a href="/offices"><button type="button" className="btn btn-trans">View Political Offices</button></a>
          </div> 
          <div className="card">
              <a href="#"><button type="button" className="btn btn-trans">View All your Votes</button></a>
          </div> 
          <div className="card">
              <a href="#"><button type="button" className="btn btn-trans">View Election Results</button></a>
          </div> 
        </div>
      </div>
    )
  }
}

export default Showcase;
