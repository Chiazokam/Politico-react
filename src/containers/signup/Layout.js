import React from 'react';
import { Header, Footer, PlainStrip, PageContainer } from '../../components/global';
import { Showcase } from '../../components/signup';

const Layout = () => {
  const navItems = ['home', 'about', 'login'];
  return (
    <React.Fragment>
      <Header navItems={navItems}/>
      <PlainStrip />
      <Showcase />
      <Footer />
    </React.Fragment>
    )
}

export default Layout;
