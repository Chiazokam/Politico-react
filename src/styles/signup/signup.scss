
.main-signup{
  border-radius: 4px;
  width: 45%;
  height: auto;
  text-align: left;
  border: 1px solid #2E2633;
  box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
  background-color: #2E2633;
  color: #DCE9BE;
  margin: 3% auto;
 }

.btn-signup{
  margin: 0 auto;
  width: 90%;
}

@media(max-width:768px){
  .main-signup{
    width: 90%;
    margin: 25px auto;
    text-align: center;
    border: 1px solid #2E2633;
    box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
   }
  
  form{
    padding: 40px 0 20px 0;
  }
  
  .form-field input{
    min-width: 100%;
  }
  
  .form-field textarea{
    min-width: 100%;
  }
}
