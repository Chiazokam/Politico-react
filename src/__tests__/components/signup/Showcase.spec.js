import React from 'react';
import "@babel/polyfill";
import { shallow } from 'enzyme';
import { SignupShowcase } from '../../../components/signup/Showcase';

describe('<Signup Showcase />', () => {
  it('renders correctly', () => {
    const props = {
      errors: {
        message: 'Wrong input'
      },
      signupUser: jest.fn()
    };
    const tree = shallow(<SignupShowcase {...props} />);
    expect(tree).toMatchSnapshot(); 
  });
});
