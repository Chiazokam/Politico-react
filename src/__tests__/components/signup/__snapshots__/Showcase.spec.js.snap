// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup Showcase /> renders correctly 1`] = `ShallowWrapper {}`;
